Source/Core/Core/PowerPC/Tables/instructions.csv

```diff
@@ -0,0 +1,637 @@
+stfs;StoreFP;FL_IN_FLOAT_S | FL_IN_A0 | FL_USE_FPU | FL_LOADSTORE
+stfsu;StoreFP;FL_IN_FLOAT_S | FL_OUT_A | FL_IN_A | FL_USE_FPU | FL_LOADSTORE
+stfd;StoreFP;FL_IN_FLOAT_S | FL_IN_A0 | FL_USE_FPU | FL_LOADSTORE
+stfdu;StoreFP;FL_IN_FLOAT_S | FL_OUT_A | FL_IN_A | FL_USE_FPU | FL_LOADSTORE
+psq_l;LoadPS;FL_OUT_FLOAT_D | FL_IN_A0 | FL_USE_FPU | FL_LOADSTORE
+psq_lu;LoadPS;FL_OUT_FLOAT_D | FL_OUT_A | FL_IN_A | FL_USE_FPU | FL_LOADSTORE
+
+->;OpId59
+psq_st;StorePS;FL_IN_FLOAT_S | FL_IN_A0 | FL_USE_FPU | FL_LOADSTORE
+psq_stu;StorePS;FL_IN_FLOAT_S | FL_OUT_A | FL_IN_A | FL_USE_FPU | FL_LOADSTORE
+
+->;OpId63
+
+===;OpId4;1;5;OPCD4 SubOps
+->;PSCompare
+
+
+
+
+
+->;PSLoad
+->;PSStore
+->;PSSign
+
+ps_sum0;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_sum1;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_muls0;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_muls1;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_madds0;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_madds1;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+->;PSMerge
+
+ps_div;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AB | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF;;17
+
+ps_sub;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AB | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_add;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AB | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+dcbz_l;System;FL_IN_A0B | FL_LOADSTORE
+ps_sel;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU
+ps_res;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_B | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_mul;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_AC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_rsqrte;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_B | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF;;2
+
+ps_msub;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_madd;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_nmsub;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+ps_nmadd;PS;FL_OUT_FLOAT_D | FL_IN_FLOAT_ABC | FL_RC_BIT_F | FL_USE_FPU | FL_SET_FPRF
+
+===;PSCompare;6;5;Paired single compare
+ps_cmpu0;PS;FL_IN_FLOAT_AB | FL_SET_CRn | FL_USE_FPU | FL_READ_FPRF | FL_SET_FPRF
+ps_cmpo0;PS;FL_IN_FLOAT_AB | FL_SET_CRn | FL_USE_FPU | FL_READ_FPRF | FL_SET_FPRF
+ps_cmpu1;PS;FL_IN_FLOAT_AB | FL_SET_CRn | FL_USE_FPU | FL_READ_FPRF | FL_SET_FPRF
+ps_cmpo1;PS;FL_IN_FLOAT_AB | FL_SET_CRn | FL_USE_FPU | FL_READ_FPRF | FL_SET_FPRF
+
+===;PSLoad;6;1;Paired single load
+psq_lx;LoadPS;FL_OUT_FLOAT_D | FL_IN_A0B | FL_USE_FPU | FL_LOADSTORE
+psq_lux;LoadPS;FL_OUT_FLOAT_D | FL_OUT_A | FL_IN_AB | FL_USE_FPU | FL_LOADSTORE
+===;PSStore;6;1;Paired single store
+psq_stx;StorePS;FL_IN_FLOAT_S | FL_IN_A0B | FL_USE_FPU | FL_LOADSTORE
+psq_stux;StorePS;FL_IN_FLOAT_S | FL_OUT_A | FL_IN_AB | FL_USE_FPU | FL_LOADSTORE
+
```

Comment:
This section is length-1, shouldn't it only have two lines?

Reply:
But it does … as described at the top, empty lines at the end of a table are ignored. This empty line is there mostly for readability.